### DISCUSSION TITLE
Client commands - dispatcher is null

### DISCUSSION THREAD

**Discussion by skyeredwood** [FabricMC/fabric-api#2489] [Mod Dev Support] [answered]
I can't find the exact stack trace, not at my PC currently. However, as part of my mod I would like to register some client-side commands. Past mods I've made have used v1 of the client command API, however since then the API seems to have moved to v2, which has a slightly different syntax. 

To cut a long story short, this means I cannot register commands unless there is an active connection to a server. If there isn't, `ClientCommandManager#getActiveDispatcher()` will return null. I can't find a way to run it post-server connect: I've tried mixing in to `MinecraftClient`, `ConnectScreen`, and as the server the mod is made for loads a resource pack I've tried to use a `ResourceReloader` to run the task after the pack is loaded (although I'm still not sure that would work, and I can't find a way to register the reloader with the `ResourceManager`).

How should I go about registering client-side only commands?

**Comment by haykam821** [comment, marked as answer]
Use the `ClientCommandRegistrationCallback.EVENT` event to handle client command registration. The handler will be passed a non-null instance of `CommandDispatcher<FabricClientCommandSource>`.

**Reply by skyeredwood** [reply]
Oh, didn't see that. I mixed into the `ClientPlayNetworkHandler` instead - if that doesn't work, I'll use that. Thanks a bunch.
